src/reference/snakemake-style-guide.rst

```diff
@@ -46,3 +46,143 @@ Avoid the ``message`` rule attribute
+   more complicated things than you can with direct interpolation, such
+   as list comprehensions.
+
+-  Snakemake can automatically discover which rules have parameter
+   values that are different than the last run and show what output
+   files are affected (``--list-params-changes``).
+
+Always use quoted (:q) interpolations
+=====================================
+
+When building shell commands to run, Snakemake does not by default
+properly quote interpolated values. This works fine if the interpolated
+value doesn't contain spaces or other special shell metacharacters (like
+quotes or backslashes), but it is fragile and a time-bomb waiting to
+break on future values.
+
+Standard best practice in any language or environment is to always quote
+parameters in generated shell commands. Snakemake supports this using
+the ``:q`` modifier for interpolation:
+
+::
+
+   params:
+       file = "filename with spaces.txt"
+   shell:
+       "wc -l {params.file:q}"
+
+Not quoting these values is also a security risk.
+
+It may be tempting to make an exception for parameters with multiple
+values where you want each become a separate command-line argument, such
+as a parameter listing three filenames. In this case, however, it's
+recommended that you make the parameter a list instead of a single
+string. Snakemake will interpolate it correctly:
+
+::
+
+   params:
+       files = ["a.txt", "b.txt", "c.txt"]
+   shell:
+       "wc -l {params.files:q}"
+
+Use triple-quoted command definitions
+=====================================
+
+Using triple-quoted (``"""`` or ``'''``) command definitions make it
+much easier to build readable commands with one-option per line. It also
+avoids any nested quoting issues if you need to use literal single or
+double quotes in your command.
+
+Example:
+
+::
+
+   shell:
+       """
+       augur parse \
+           --sequences {input:q} \
+           --fields {params.fields:q} \
+           --output-sequences {output.sequences:q} \
+           --output-metadata {output.metadata:q}
+       """
+
+Provide descriptive messages for each rule
+==========================================
+
+Rules support a ``message:`` block which defines a string to print when
+the rule is run. This is very helpful for tracking build progress and in
+general describing what's happening. Note that you can use interpolation
+within the message, so you can surface important parameter values or
+conditional inputs or anything else that might be useful.
+
+Example:
+
+::
+
+   rule traits:
+       message: "Inferring ancestral traits {params.columns!s}"
+
+Always use relative paths
+=========================
+
+Relative paths (paths that don't start with ``/``) mean that anyone can
+run the build without running into portability issues caused by paths
+specific to your computer.
+
+Our convention for builds is that Nextstrain components are available
+locally if needed as sibling directories, for example ``../fauna/`` and
+``../auspice/``.
```

Comment:
This ¶ should probably be dropped given that most builds don't/won't use Fauna and Auspice is available as the `auspice` command now.

Reply:
Done: 295cbb7abc237939821bc8c9bb291468ed2a197a